### DISCUSSION TITLE
Discussions vs Issues

### DISCUSSION THREAD

**Discussion by frenck** [home-assistant/architecture#447] [General] [answered]
GitHub gave our project the possibility to test and use the new GitHub discussions feature.

We are looking at trying to use it at the `frontend` and this `architecture` repository.

**The question now becomes, should we use it or not?**

So Discussions are more aimed at... yes, you guessed it, discussions and has some interesting features, but also some missing features compared to issues.

## Pros

- Threaded discussions/replies in a topic (easier to follow).
- Clear separation from an "issue".
- The final answer can be marked as re-"solving" answer.
- Issues can be moved/converted to a discussion.
- When creating an issue, one of the template/issue choices can be a discussion starter.
- Discussions are categorized.
- PRs can link to discussions.

## Cons

- Discussions don't have labels like issues.
- A discussion cannot be converted to an issue, however, an issue can be created that references an discussion with a single click.
- Dicussions cannot be closed like issues, however, they can be resolved or locked.
- No templates. There are no predefined "issue" templates, so the creator of a discussion always starts with a blank slate/free-form.
- PRs cannot resolve/close/fix an discussion.
- Linked PRs are not visible from the discussion.

## Moving forward

If we are doing this, cleaning up the architecture repository might be a good moment as well.

Please leave a 👍 or 👎 as a emoji reaction to express your opinion on this matter or post a reply with additional ideas, context, adjustments or 🤷

**Comment by frenck** [comment]
Now I wonder, can a discussion be "resolved" using a PR?

**Reply by frenck** [reply]
Did a test in #448 

Unfortunately, it doesn't pop up or resolve the discussion.

Updated topic with finding.

**Reply by balloob** [reply]
But we can add a reply with a link to the PR and mark that as the answer.

**Reply by frenck** [reply]
That is true 👍

**Comment by Kane610** [comment]
I do like the frontends use of feature requests that way, would be awesome if that could be utilised for core as well rather than the hard to track forum category

**Reply by frenck** [reply]
Yes, that would be the use case we are going to explore on the frontend repository.

https://github.com/home-assistant/frontend/discussions

**Reply by frenck** [reply]
I guess for most repositories this could work for feature requests. The general concern:

- Accessibility for the end-user. A large part of our user base is scared (assumption) for GitHub (which makes sense, feels technical).
- For the core it would be quite challenging. For example, requests for new integrations, is that something you'd want to track on the core repository?

**Reply by Kane610** [reply]
So maybe not replace but complement at the very least?

**Reply by frenck** [reply]
Dunno? Maybe it doesn't matter at all?
I was talking to @bramkragten, he suggested to just make a category for new integrations or device support.

At least it allows us to streamline everything. Which I kinda prefer personally to achieve in the long term.

**Reply by MartinHjelmare** [reply]
Is it possible to not be notified of discussions in a repo but still track other things? I'd be hesitant to increase the notification burden of the core repository without filtering options.

**Reply by frenck** [reply]
It is considered the same level as an issue at this point. You can, however, unsub from a topic.
Would be a good feedback thing to send out to GitHub.

**Comment by frenck** [comment, marked as answer]
Alright, let's take this route! 👍

**Comment by nareddyt** [comment]
A quick observation: When moving an issue to a discussion, it closes the original issue. This is misleading if you see this via an Github email notification or via the official Github app - it just says the issue has been closed. Only the web UI shows the issue has been moved to discussions.

Obviously not your fault, but I suggest adding a comment before moving issues to prevent confusion. Otherwise it seems like the issue was closed for no reason. Hopefully Github will standardize the UI across platforms once discussions is out of beta.

**Reply by frenck** [reply]
Discussions is out of beta.

I'll leave a note 👍
